### PR TITLE
[Issue #387] Cooldown Tracker should not include boss abilities

### PR DESCRIPTION
**Pull request by sMteX**
**Original issue**: #387 

Not sure if there is some general solution to boss abilities, because even in the mentioned example, the casts are of Astral Vulnerability and it's from a player to a friendly player, so there's no way of knowing if that's a boss specific mechanic (just by looking at the event object). 

Instead I added a simple static array that contains spell IDs you wish to ignore. This can be furthermore extended by spec CooldownTrackers to add additional ignored spells (I suggested that in PR #405 and eventually rewrote that in this PR - works completely the same), for example see Marksmanship Hunter Cooldown Tracker.

Before: 
cooldowns-before

After:
cooldowns-after

### CODE REVIEW

**Comment by kfinch** [comment]
I'm pretty sure you have write access, I'll let you do the update/merge when you're ready.

**Comment by anom0ly** [comment]
Sorry for coming late to the game on this one.

Mistweavers make use of similar functionality. Can you make the same adjustments that were made to MM to the MW module?

Thanks.

File: src/Parser/MarksmanshipHunter/Modules/Features/CooldownTracker.js [hunk 1]
```diff
@@ -15,18 +13,10 @@ class CooldownTracker extends CoreCooldownTracker {
     },
   ];
 
-  castEventSpells = [
+  static ignoredSpells = [
+    ...CooldownTracker.ignoredSpells,
```
Comment (anom0ly) [inline src/Parser/MarksmanshipHunter/Modules/Features/CooldownTracker.js]: Can you extend these same changes to the Mistweaver Module also? Thanks.

**Comment by MartijnHols** [comment]
http://www.wowhead.com/spell=207630/annihilation as well

File: src/Parser/MarksmanshipHunter/Modules/Features/CooldownTracker.js (hunk 1 above)
Comment (MartijnHols) [reply src/Parser/MarksmanshipHunter/Modules/Features/CooldownTracker.js]: Wouldn't it make sense to do this in the core?

Comment (anom0ly) [reply src/Parser/MarksmanshipHunter/Modules/Features/CooldownTracker.js]: I believe my comment was not clear. MW make use of a similar function to remove the extra cast triggers for Chi Burst and Refreshing Jade Wind, for example. So to keep it consistent just also make those changes to the MW `CooldownTracker` module. I didn't mean to extend the core boss ability clean up to Mistweavers. I agree that should be done in the core parser.

**Comment by sMteX** [comment]
I couldn't find any sensible log when there was a cast event of Annihilation during a cooldown, but looking at the Events tab, I looked for Annihilation cast and added that so I hope it's okay.

**Comment by anom0ly** [comment]
Looks good from my end. Just need to aware this would need to be monitored tier by tier.

**Pull request merged**